lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -178,7 +178,7 @@ <h3 class="course-title">
               </a>
               <div class="actions-dropdown" id="actions-dropdown-${dashboard_index}" aria-label="${_('Additional Actions Menu')}">
                 <ul class="actions-dropdown-list" id="actions-dropdown-list-${dashboard_index}" aria-label="${_('Available Actions')}" role="menu">
-                  % if cert_status.get('status') not in DISABLE_UNENROLL_CERT_STATES:
+                  % if cert_status is None or cert_status.get('status') not in DISABLE_UNENROLL_CERT_STATES:
```

Comment:
I would much rather see a single Boolean passed that determines if this component should be shown. We should not have Python in template code. Doing so allows the very big you're fixing to exist in the fist place.